ext/standard/string.c

```diff
@@ -1899,11 +1899,14 @@ PHP_FUNCTION(strpos)
 		RETURN_FALSE;
 	}
 
-	if (!ZSTR_LEN(needle)) {
-		php_error_docref(NULL, E_WARNING, "Empty needle");
+	if (ZSTR_LEN(haystack) == 0 || ZSTR_LEN(needle) > ZSTR_LEN(haystack)) {
```

Comment:
Why this change? This will return the wrong result for `strpos("", "")`, which should be 0, not false.

Reply:
The stripos function did the following checks:
```c
	if (ZSTR_LEN(needle) == 0 || ZSTR_LEN(needle) > ZSTR_LEN(haystack)) {
		RETURN_FALSE;
	}
```
And
```
if (ZSTR_LEN(haystack) == 0) {
		RETURN_FALSE;
}
```
Which ``strpos`` did not that's why I added this check such that these two function behave similarly, or am I missing something else?